cassandra/cqlengine/columns.py

```diff
@@ -435,10 +437,20 @@ def to_python(self, value):
             return value
         elif isinstance(value, date):
             return datetime(*(value.timetuple()[:6]))
+        elif isinstance(value, six.string_types):
+            try:
+                return iso8601.parse_date(value)
+            except iso8601.ParseError:
+                pass
+
         try:
             return datetime.utcfromtimestamp(value)
         except TypeError:
-            return datetime.utcfromtimestamp(DateType.deserialize(value))
+            try:
+                return DateType.deserialize(value, 0)
```

Comment:
I can't figure out why this was added.
https://github.com/datastax/python-driver/commit/70d866018d3fc56e960d0f7746a5a6d33f8aadd8
I don't know why we would ever be working with serialized values in the mapping layer.
There is no test covering this line. I'm leaning toward removing it and letting a TypeError propagate in this case.
(lmkwyt)